packages/@aws-cdk/core-synthesizer/lib/staging-stack.ts

```diff
@@ -0,0 +1,152 @@
+import * as ecr from '@aws-cdk/aws-ecr';
+import * as iam from '@aws-cdk/aws-iam';
+import * as s3 from '@aws-cdk/aws-s3';
+import { DockerImageAssetSource, RemovalPolicy, Stack, StackProps } from '@aws-cdk/core';
+import { Construct, IConstruct } from 'constructs';
+
+/**
+ * Information on how a Staging Stack should look.
+ */
+export interface IStagingStack extends IConstruct {
+  /**
+   * The app-scoped, environment-keyed bucket created in this staging stack.
+   */
+  readonly stagingBucket: s3.Bucket;
+
+  /**
+   * The well-known name of the app-scpoed, environment-keyed bucket created in this staging stack.
+   */
+  readonly stagingBucketName: string;
+
+  /**
+   * The app-scoped, environment-keyed repositories created in this staging stack.
+   * A repository is created per image asset family.
+   */
+  readonly stagingRepos: Record<string, ecr.Repository>;
+
+  /**
+   * The well-known name of the IAM role assumed for publishing to s3.
+   */
+  readonly fileAssetPublishingRoleArn: string;
+
+  /**
+   * The well-known name of the IAM role assumed for publishing to ecr.
+   */
+  readonly dockerAssetPublishingRoleArn: string;
```

Comment:
For all of these I would ask: are they necessary to be exposed publicly? Can we make the API as narrow as we need it to be?

Reply:
I will refer again to [Tell, don't ask](https://martinfowler.com/bliki/TellDontAsk.html): the synthesizer could say to the stack, for every asset: "here's an asset, tell me where you'd like it to end up and what role to use", and stick that into the manifest.

That puts the staging stack fully in control of asset location, and if it wanted to it could even do crazy things like make a new bucket for every asset, or put all ECR images into the same repository, or send them off to a different account, or whatever.

This is ~ what the synthesizer does today already, but I think the point will be that this staging stack will be easier to override for users.

Reply:
We can! Does this mean that we want to move the entirety of `assetManifest` from the synthesizer to the staging stack?